### DISCUSSION TITLE
[Medium/Swift] 1347 시간 차이 비교

### DISCUSSION THREAD

**Discussion by Lautner-kwangho** [I-JUNNYLAND-I/algorithm#362] [Q&A] [answered]
안녕하세요 
다음과 같이 Medium 문제를 풀다가 시간초과가 나왔어요

제가 처음 고차함수를 사용해서 풀었더니 시간초과가 나와서
단순 for 반복문과 대체로 변경했어요

근데 시간초과가 나왔는데
Hash로 푸니까 시간초과가 나오지 않더라구요

그 이유가 궁금해요

Xcode 상에서는 처음 코드가 시간 절약이 더 돼요

### - 시간 초과 코드
``` Swift
    func minSteps(_ s: String, _ t: String) -> Int {

    var tempT = Array(t)
    var count = 0
    
    for i in s {
        for j in tempT.enumerated() {
            if i == j.element {
                count += 1
                tempT[j.offset] = Character(".")
                break
            }
        }
    }
    
    return t.count - count

    }
```
![image](https://github.com/I-JUNNYLAND-I/algorithm/assets/80211277/724bcb14-caee-4f54-b55d-3d64d829d683)

### - 시간 통과
``` Swift 
func minSteps(_ s: String, _ t: String) -> Int {
    var hash = [Character: Int]()
    
    for i in t {
        if let map = hash[i] {
            hash[i] = map + 1
        } else {
            hash[i] = 1
        }
    }
    
    var changes = 0
    for i in s {
        if let map = hash[i] {
            let newSum = map - 1
            hash[i] = newSum > 0 ? newSum : nil
        } else {
            changes += 1
        }
    }
    
    return changes

    }
```
![image](https://github.com/I-JUNNYLAND-I/algorithm/assets/80211277/8408fab6-c200-45af-948c-1df2c2fe60f0)

**Comment by Junnyjun** [comment, marked as answer]
자바의 경우 재귀가 12개? 였을시 문제가 발생하는 이슈가 잇엇습니다

이와 같은 이슈가 스위프트쪽에도 있엇을수도 있을거같아요,,

지금 사용하시는 버전과 릿코드의 스위프트 버전이 달라 생긴 이슈같네요
지금 사용하시는 스위프트 버전은 개선이 되엇나봐요

아래는 자바에서 발생하는 이슈를 첨부해드렷습니다
https://thebook.io/080337/0168/

스위프트도 따로 GC가 동작하는 언어인지는 모르겟지만 그렇다면 가능성있어보이는 블로그도 첨부해드렸습니다
https://sunny-maneg.tistory.com/entry/swift-%EC%9E%AC%EA%B7%80%ED%95%A8%EC%88%98
